### ISSUE TITLE
broken a11y types on v0.3.0

### ISSUE THREAD

**Issue by efstathiosntonas** [enzomanuelmangano/pressto#11]
Hi Enzo, hope you are doing well. 

After upgrading to 0.3.0 I cannot pass a11y props any more, i got typescript error: 

```
{
  children: Element;
  accessibilityHint: string;
  accessibilityLabel: string;
  accessibilityRole: string;
  onPress: (() => void) | undefined;
  style: {
    alignItems: "center";
    justifyContent: "center";
    paddingHorizontal: number;
    paddingVertical: number;
  };
}
is not assignable to type IntrinsicAttributes & CustomPressableProps
Property accessibilityHint does not exist on type IntrinsicAttributes & CustomPressableProps
```

```tsx
const DismissButton = createAnimatedPressable((progress: SharedValue<number>) => {
  'worklet';
  return {
    backgroundColor: interpolateColor(progress.get(), [0, 1], ['#d5534a00', '#e14f4482']),
    borderRadius: 8,
    borderCurve: 'continuous'
  };
});
```

```tsx
    <DismissButton
                  accessibilityHint={t`Close this dialog without making changes`}
                  accessibilityLabel={t`Dismiss`}
                  accessibilityRole="button"
                  onPress={onDismiss}
                  style={styles.button}
                >
                  <Text style={styles.buttonText}>{t`Dismiss`}</Text>
                </DismissButton>
```

<img width="1010" height="559" alt="Image" src="https://github.com/user-attachments/assets/9ddfe6a0-aedb-4803-ba9c-61996e1cfc2b" />
